server/src/main/java/online/dipa/hub/services/TimelineService.java

```diff
@@ -83,25 +102,219 @@ private void initializeTimelines() {
     private void initializeMilestones(final Long timelineId) {
         TimelineState sessionTimeline = findTimelineState(timelineId);
 
+        final ProjectApproachEntity projectApproach = findProjectApproach(timelineId);
+
         if (sessionTimeline.getMilestones() == null) {
-            sessionTimeline.setMilestones(this.loadMilestones(timelineId));
+            if (projectApproach.isIterative()) {
+                initializeIncrements(timelineId);
+                sessionTimeline.setMilestones(this.loadMilestones(timelineId, 1));
+            }
+            else {
+                sessionTimeline.setMilestones(this.loadMilestones(timelineId, 0));
+            }
         }
     }
 
-    private List<Milestone> loadMilestones(final Long timelineId) {
-        final ProjectTypeEntity projectTypeEntity = projectTypeRepository.findById(timelineId)
-                .orElseThrow(() -> new EntityNotFoundException(
-                        String.format(
-                                "Timeline with id: %1$s not found.",
-                                timelineId)));
+    private List<Milestone> loadMilestones(final Long timelineId, final int incrementCount) {
+
+        List<Milestone> milestones = getMilestonesFromRespository(timelineId);
+
+        if (incrementCount == 0) {
+            return milestones;
+        } 
+        else {
+            List<Milestone> incrementMilestones = new ArrayList<Milestone>();
+
+            incrementMilestones.add(milestones.get(0));
+            incrementMilestones.add(milestones.get(milestones.size() - 1));
 
-        return projectTypeEntity.getMilestones()
+            for (Entry<Increment, List<Milestone>> entry : getIncrementMilestones(milestones, timelineId, incrementCount).entrySet()) {
+                List<Milestone> milestonesList = entry.getValue();
+                incrementMilestones.addAll(milestonesList);
+            }
+
+            return incrementMilestones;
+        }
+    }
+
+    private HashMap<Increment, List<Milestone>> getIncrementMilestones(List<Milestone> initMilestones,
+            final Long timelineId,
+            final int incrementCount) {
+
+        LocalDate firstDatePeriod = initMilestones.stream().map(Milestone::getDate).min(LocalDate::compareTo).get();
+        LocalDate lastDatePeriod = initMilestones.stream().map(Milestone::getDate).max(LocalDate::compareTo).get();
+
+        List<Increment> incrementsList = loadIncrements(timelineId, incrementCount);
+
+        List<Milestone> templateMilestones = initMilestones;
+        templateMilestones.remove(templateMilestones.size() - 1);
+        templateMilestones.remove(0);
```

Comment:
Local variable 'templateMilestones' is redundant:

initMilestones.remove(initMilestones.size() - 1);
initMilestones.remove(0);